testdata/instrumenter/BinaryExpr.gobco

```diff
@@ -27,84 +27,66 @@ func binaryExpr(i int, a bool, b bool, c bool) {
 	//
 	// Also, gobco only looks at the parse tree without any type resolution.
 	// Therefore it cannot decide whether a variable is boolean or not.
-	both := GobcoCover(2, a) && GobcoCover(3, b)
-	either := GobcoCover(4, a) || GobcoCover(5, b)
+	both := GobcoCover(2, a && b)
+	either := GobcoCover(3, a || b)
 	_, _ = both, either
 
 	// When a long chain of '&&' or '||' is parsed, it is split into
 	// the rightmost operand and the rest, instrumenting both these
 	// parts.
-	_ = GobcoCover(6, i == 11) ||
-		GobcoCover(7, i == 12) ||
-		GobcoCover(8, i == 13) ||
-		GobcoCover(9, i == 14) ||
-		GobcoCover(10, i == 15)
-	_ = GobcoCover(11, i != 21) &&
-		GobcoCover(12, i != 22) &&
-		GobcoCover(13, i != 23) &&
-		GobcoCover(14, i != 24) &&
-		GobcoCover(15, i != 25)
+	_ = GobcoCover(4, i == 11 ||
+		i == 12 ||
+		i == 13 ||
+		i == 14 ||
+		i == 15)
+
+	_ = GobcoCover(5, i != 21 &&
+		i != 22 &&
+		i != 23 &&
+		i != 24 &&
+		i != 25)
 
 	// The operators '&&' and '||' can be mixed as well.
-	_ = GobcoCover(16, i == 31) ||
-		GobcoCover(17, i >= 32) && GobcoCover(18, i <= 33) ||
-		GobcoCover(19, i >= 34) && GobcoCover(20, i <= 35)
+	_ = GobcoCover(6, i == 31 ||
+		i >= 32 && i <= 33 ||
+		i >= 34 && i <= 35)
 
 	m := map[bool]int{}
-	_ = GobcoCover(21, m[GobcoCover(22, i == 41)] == m[GobcoCover(23, i == 42)])
+	_ = GobcoCover(7, m[GobcoCover(8, i == 41)] == m[GobcoCover(9, i == 42)])
 
 	// In complex conditions, only instrument the terminal conditions
```

Comment:
This comment and the other comments would need to be updated to reflect the changed behavior.

Reply:
I will fix this when I separating test suite.